File: pset2/readability/readability.c
# include <stdio.h>
# include <cs50.h>
# include <string.h>
# include <ctype.h>
# include <math.h>

int letters, sentences, x;
//set initial grade level
int index;

int main(void)
{
    //Get input text from the user
    string text = get_string("Text: ");
    
    //count for letters, words, and sentences in the text
    int n = strlen(text);
    int words = 1;
    
    for (int i = 0; i < n; i++)
    {
        if (isalnum(text[i]) || islower(text[i]) || isupper(text[i]))
        {
            letters++;
        }
        else if (isspace(text[i]))
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
        
    }
   // printf("%i, %i, %i\n", letters, words, sentences);

    // calculate coleman liau index
    index = round(0.0588 * ((letters * 100) / (float) words) - 0.296 * ((sentences * 100) / (float)words) - 15.8);
    

    //grade level
    x = index;
    printf("%i\n", x);

    //TODO: print the grade level of the text
    if (x >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (x <= 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", x);
    }

   
}

File: pset6/readability.py
import importlib


# prompt user for text
text = input("Text: ")
TextLength = len(text)
# initialise parameters
sentences =letters = 0
words = 1
# count for letters, words, and sentences in the text
for i in range(TextLength):
    if (text[i].isalpha()):
        letters += 1
    elif (text[i].isspace()):
        words += 1
    elif (text[i] == "." or text[i] == "?" or text[i] == "!"):
        sentences += 1
print(letters)
print(words)
print(sentences)
# calculate coleman liau index
L = (letters / words) * 100
S = (sentences / words) * 100
index = round(0.0588 * L - 0.296 * S - 15.8)
if (index >= 16):
    print("Grade 16+")
elif (index <=1):
    print("Before Grade 1")
else:
    print(f"Grade {index}")



